astropy/utils/iers/iers.py

```diff
@@ -802,3 +825,301 @@ def _substitute_iers_b(cls, table):
+        """Create a table from a file like the IETF ``leap-seconds.list``.
+
+        Parameters
+        ----------
+        file : path, optional
+            Full local or network path to the file holding leap-second data
+            in a format consistent with that used by IETF.  Up to date versions
+            can be retrieved from `astropy.utils.iers.IETF_LEAP_SECOND_URL`.
+
+        Notes
+        -----
+        The file *must* contain the expiration date in a comment line, like
+        '# File expires on:  28 June 2020'
+        """
+        names = ['ntp_seconds', 'tai_utc', 'comment', 'day', 'month', 'year']
+        # Note: ntp_seconds does not fit in 32 bit, so causes problems on
+        # 32-bit systems without the np.int64 converter.
+        self = cls._read_leap_seconds(
+            file, names=names, include_names=names[:2],
+            converters={'col1': [convert_numpy(np.int64)]})
+        self['mjd'] = (self['ntp_seconds']/86400 + 15020).round()
+        # Note: cannot use Time in this routine.
+        dt = [datetime(1900, 1, 1) + timedelta(mjd-15020)
+              for mjd in self['mjd']]
+        self['day'] = [d.day for d in dt]
+        self['month'] = [d.month for d in dt]
+        self['year'] = [d.year for d in dt]
+        return self
+
+    @classmethod
+    def from_erfa(cls, built_in=False):
+        """Create table from the leap-second list in ERFA.
+
+        Parameters
+        ----------
+        built_in : bool
+            If `False` (default), retrieve the list currently used by ERFA,
+            which may have been updated.  If `True`, retrieve the list shipped
+            with erfa.
+        """
+        current = cls(erfa.leap_seconds.get())
+        current._expires = erfa.leap_seconds.expires
+        if not built_in:
+            return current
+
+        try:
+            erfa.leap_seconds.set(None)  # reset to defaults
+            return cls.from_erfa(built_in=False)
+        finally:
+            erfa.leap_seconds.set(current)
+
+    def update_erfa_leap_seconds(self, initialize_erfa=False):
+        """Add any leap seconds not already present to the ERFA table.
+
+        This method matches leap seconds with those present in the ERFA table,
+        and extends the latter as necessary.
+
+        Parameters
+        ----------
+        initialize_erfa : bool, or 'only', or 'empty'
+            Initialize the ERFA leap second table to its built-in value before
+            trying to expand it.  This is generally not needed but can help
+            in case it somehow got corrupted.  If equal to 'only', the ERFA
+            table is reinitialized and no attempt it made to update it.
+            If 'empty', the leap second table is emptied before updating, i.e.,
+            it is overwritten altogether (note that this may break things in
+            surprising ways, as most leap second tables do not include pre-1970
+            pseudo leap-seconds; you were warned).
```

Comment:
I think this is reasonable, including the warning, but what I really wanted instead of `empty` is a way to provide a way to do something like  "give me all the leap seconds up to a certain point, but then not the most recent one".  

But now that I say it, I think that can now be achieved like so:

```
ls_tab = LeapSeconds.from_erfa()
del ls_tab[-1] # or whichever row I'm trying to get rid of...
ls_tab. update_erfa_leap_seconds('empty')
```
right?  I.e., that should then still keep all of the pre-1970 leap seconds?

Assuming @mhvk agrees that works the way I'm thinking I think this is good.

Reply:
Yes, that would be the trick.

Reply:
OK, then I'll plan to write a follow-on PR (which can go after this is merged) that adds a short example like this to the docs along with a what's new section.  But doesn't need to be in this PR.

Reply:
Yes, extra docs would be very nice but are not essential for the auto-update part.